runtime/parachains/src/inclusion/mod.rs

```diff
@@ -179,6 +215,38 @@ pub fn minimum_backing_votes(n_validators: usize) -> usize {
 	sp_std::cmp::min(n_validators, 2)
 }
 
+/// Provides introspection into the UMP dispatch queue of a parachain.
+pub trait UmpQueueTracker {
+	fn message_count(para: ParaId) -> u64;
+}
+
+impl UmpQueueTracker for () {
+	fn message_count(_: ParaId) -> u64 {
+		0
+	}
+}
+
+/// Aggregate message origin for the `MessageQueue` pallet.
+///
+/// Can be extended to serve further use-cases besides just UMP. Changing this possibly requires a migration.
```

Comment:
Would it? Let's fix the indices here anyway. If we only add items then it should be fine.